core/variant/array.h

```diff
@@ -135,6 +135,7 @@ class Array {
 	_FORCE_INLINE_ void append(const Variant &p_value) { push_back(p_value); } //for python compatibility
 	void append_array(const Array &p_array);
 	Error resize(int p_new_size);
+	Error reserve(int64_t p_new_size);
```

Comment:
My only concern here is that the reserve function supports a wider range than the resize, for uniformity I'd suggest using `int` to not create strange cases, you could easily have cases where you do `reserve(size)` followed by operations adding elements or resizing triggering errors or weirdness as the size isn't supported